### PR TITLE
Migrate to Kotlin DSL

### PR DESCRIPTION
**Pull request by johnshea**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("com.android.application")
+    id("org.jetbrains.kotlin.android")
+    id("com.google.devtools.ksp") version "1.8.21-1.0.11"
```
Comment (jhansitavva) [inline app/build.gradle.kts]: Could you please check if we can upgrade this to the latest version?

File: app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        compose = true
+    }
+    composeOptions {
+        kotlinCompilerExtensionVersion = rootProject.extra["compose_compiler_version"].toString()
+    }
+}
+
+dependencies {
+
+    implementation(platform("androidx.compose:compose-bom:2023.06.00"))
```
Comment (jhansitavva) [inline app/build.gradle.kts]: Nit: Latest Stable: 2023.06.01

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #Thu Jun 01 13:39:29 PDT 2023
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
```
Comment (jhansitavva) [inline gradle/wrapper/gradle-wrapper.properties]: Nit: 8.2

File: app/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,77 @@
+/*
```
Comment (jhansitavva) [inline app/build.gradle.kts:1]: Consider adding
`@file:Suppress("UnstableApiUsage")
`
?

**Review by jhansitavva — APPROVED** [review]
Minor comments regarding the dependencies version. 
One ask is let's be consistent with suppressing the `UnstableApiUsage`
`@file:Suppress("UnstableApiUsage")`

File: app/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        targetCompatibility = JavaVersion.VERSION_17
+    }
+    kotlinOptions {
+        jvmTarget = "17"
+    }
+    buildFeatures {
+        compose = true
+    }
+    composeOptions {
+        kotlinCompilerExtensionVersion = rootProject.extra["compose_compiler_version"].toString()
```
Comment (jhansitavva) [inline app/build.gradle.kts]: Let us be consistent with this usage. 
Either set it here or use a variable, both are fine!

File: build.gradle.kts [hunk 6]
```diff
@@ -15,15 +15,15 @@
  */
 
 buildscript {
-    ext {
-        arch_lifecycle_version = '2.6.1'
-        compose_compiler_version = '1.4.7'
-        room_version = '2.5.1'
+    extra.apply {
+        set("arch_lifecycle_version", "2.6.1")
+        set("compose_compiler_version", "1.4.7")
```
Comment (jhansitavva) [inline build.gradle.kts]: Update the compiler version `1.5.0`

File: app/build.gradle.kts (hunk 1 above)
Comment (johnshea) [reply app/build.gradle.kts]: Upgraded to latest version.

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (johnshea) [reply gradle/wrapper/gradle-wrapper.properties]: Upgraded to latest version.

File: app/build.gradle.kts (hunk 5 above)
Comment (johnshea) [reply app/build.gradle.kts]: Updated code to set it here.

File: build.gradle.kts (hunk 6 above)
Comment (johnshea) [reply build.gradle.kts]: Upgraded to latest version.

File: app/build.gradle.kts (hunk 4 above)
Comment (johnshea) [reply app/build.gradle.kts:1]: Ack.

File: app/build.gradle.kts (hunk 2 above)
Comment (johnshea) [reply app/build.gradle.kts]: Upgraded to latest version.

**Review by jhansitavva — APPROVED** [review]

**Pull request merged**